### PR TITLE
Add implementation for these config APIs

### PR DESCRIPTION
**Pull request by abdelrady**
Add implementation for these config APIs:
* /dashboards -> Lists dashboard names
* /dashboards/{dashboard}/tabs -> Lists the dashboard tab names
* /dashboards/{dashboard} -> Returns a Dashboard config's metadata

+ Unit tests added
+ Fix 404 not found issue for dashboard groups locating (missing Normalize comparison)

### CODE REVIEW

**Comment by chases2** [comment]
Looks good so far! Looking forward to seeing the rest of the tests

**Comment by abdelrady** [comment]
Thank you @chases2 . I added the unit tests. You may take a look.

**Comment by fejta** [comment]
/hold

Once this is ready to merge please squash your commits.

File: pkg/api/v1/config.go
Comment (chases2) [inline pkg/api/v1/config.go:187]: nit: Please add a TODO here for the link (or add it in the format `/dashboards/{dashboard}/tabs/{tab}`)

**Review by chases2 — APPROVED** [review]
xref #549 and #579

This looks great! Feel free to unhold when you've squashed commits and are ready to merge. Thanks!

File: pkg/api/v1/config.go
Comment (abdelrady) [reply pkg/api/v1/config.go:187]: Link added.

**Comment by abdelrady** [comment]
@chases2 Thank you! I added the link in the tabs api, updated unit tests, and squashed the commits. I will merge once all checks are passed.

File: pkg/api/v1/config.go
Comment (fejta) [inline pkg/api/v1/config.go:128]: It looks like this writes json objects, not protos?

**Review by fejta — APPROVED** [review]

File: pkg/api/v1/config.go
Comment (abdelrady) [reply pkg/api/v1/config.go:128]: I think we just refer to the response type in the proto file.

**Review by chases2 — APPROVED** [review]

File: pkg/api/v1/config.go
Comment (fejta) [reply pkg/api/v1/config.go:128]: But it isn't a response proto, it is a response JSON

Comment (abdelrady) [reply pkg/api/v1/config.go:128]: I see. I will create another PR to resolve this in all config APIs

**Pull request merged**
